lib/glific/flows.ex

```diff
@@ -26,8 +26,60 @@ defmodule Glific.Flows do
 
   """
   @spec list_flows(map()) :: [Flow.t()]
-  def list_flows(args),
-    do: Repo.list_filter(args, Flow, &Repo.opts_with_name/2, &filter_with/2)
+  def list_flows(args) do
+    flow_revision_list = get_status_list()
+
+    Repo.list_filter(args, Flow, &Repo.opts_with_name/2, &filter_with/2)
+    |> Enum.map(fn flow ->
+      Map.merge(
+        flow,
+        flow_revision_list |> Enum.find(fn status -> Map.get(status, :id) == flow.id end)
+      )
+    end)
+  end
+
+  @spec get_published(map(), Flow.t()) :: map()
+  defp get_published(published_list, flow) do
+    checked = published_list |> Enum.find(fn status -> Map.get(status, :id) == flow.id end)
+    if is_nil(checked), do: %{}, else: checked
+  end
+
+  @spec get_status_list :: [Flow.t()]
+  defp get_status_list do
```

Comment:
Combine the below two queries into

fr.status == "published" or fr.revision_number == 0